src/lib/c-utils.h

```diff
@@ -69,9 +70,41 @@ static inline
 r_ssize r_ssize_min(r_ssize x, r_ssize y) {
   return (y < x) ? y : x;
 }
+
 static inline
 r_ssize r_ssize_max(r_ssize x, r_ssize y) {
   return (y < x) ? x : y;
 }
 
+static inline
+double r_ssize_as_double(r_ssize x) {
+  if (x > DBL_MAX || x < -DBL_MAX) {
+    r_stop_internal("r_ssize_as_double", "Result can't be represented as `double`.");
+  }
+
+  return (double) x;
+}
+
+static inline
+r_ssize r_double_as_ssize(double x) {
+  x = round(x);
```

Comment:
`round()` or `floor()`? i.e. should `0.6` round up to `1`? i think base R uses `(R_xlen_t) x` which would truncate to `0`

Reply:
I wondered about this and couldn't decide what is best.

Reply:
When this matters, it's probably best for the caller to pre-trim the double.